src/org/jitsi/impl/neomedia/codec/video/vp8/DePacketizer.java

```diff
@@ -566,6 +637,60 @@ private static int getPictureId(byte[] input, int offset)
 
         }
 
+        /**
+         * Sets the extended picture ID for the VP8 payload specified in the
+         * buffer that is passed as an argument.
+         *
+         * @param buf the byte buffer that contains the VP8 payload.
+         * @param off the offset in the byte buffer where the VP8 payload
+         * starts.
+         * @param len the length of the VP8 payload in the byte buffer.
+         * @return true if the operation succeeded, false otherwise.
+         */
+        public static boolean setExtendedPictureId(
+            byte[] buf, int off, int len, int val)
+        {
+            if (!hasExtendedPictureId(buf, off, len))
+            {
+                return false;
+            }
+
+            buf[off + 2] = (byte) (0x80 | (val >> 8) & 0x7F);
+            buf[off + 3] = (byte) (val & 0xFF);
+
+            return true;
+        }
+
+        /**
+         * Sets the TL0PICIDX field for the VP8 payload specified in the
+         * buffer that is passed as an argument.
+         *
+         * @param buf the byte buffer that contains the VP8 payload.
+         * @param off the offset in the byte buffer where the VP8 payload
+         * starts.
+         * @param len the length of the VP8 payload in the byte buffer.
+         * @return true if the operation succeeded, false otherwise.
+         */
+        public static boolean setTL0PICIDX(byte[] buf, int off, int len, int val)
+        {
+            if (!hasTL0PICIDX(buf, off, len))
+            {
+                return false;
+            }
+
+            int sz = getSize(buf, off);
+            if (hasKEYIDX(buf, off, len))
+            {
+                buf[off + sz - 2] = (byte) val;
+            }
+            else
+            {
+                buf[off + sz - 1] = (byte) val;
```

Comment:
In light of the comment above about KEYIDX, I would change this find the offset based on the presence (and length) of PictureID